### PR TITLE
[WIP] OSDOCS-2210: OSDK-1713 Custom Scorecard storage feature

### PR DESCRIPTION
**Pull request by michaelryanpeter**
Release: 4.10

[OSDOCS-2210](https://issues.redhat.com/browse/OSDOCS-2210)

Docs preview: https://deploy-preview-36751--osdocs.netlify.app/openshift-enterprise/latest/operators/operator_sdk/osdk-scorecard#osdk-scorecard-store-output_osdk-scorecard

### CODE REVIEW

File: modules/osdk-scorecard-store-output.adoc [hunk 1]
```diff
@@ -0,0 +1,24 @@
+// Module included in the following assemblies:
+//
+// * operators/operator_sdk/osdk-scorecard.adoc
+
+:osdk_ver: v1.8.0
+
+[id="osdk-scorecard-store-output_{context}"]
+= Scorecard output storage
```
Comment (jmrodri) [inline modules/osdk-scorecard-store-output.adoc:8]: Do we need to add more context about this feature?  Feels like there should be more. I'll have @theishshah look this over as well.

Comment (michaelryanpeter) [reply modules/osdk-scorecard-store-output.adoc:8]: @jmrodri @theishshah I am happy to add additional context. I was working primarily off of the [upstream docs ](https://sdk.operatorframework.io/docs/testing-operators/scorecard/custom-tests/#storing-scorecard-test-output), so please let me know what I can add.

Comment (theishshah) [reply modules/osdk-scorecard-store-output.adoc:8]: This feature can also be enabled in the config.yaml via the following:
```
tests:
  - entrypoint:
    - scorecard-test
    - <test-entrypoint>
    image: quay.io/operator-framework/scorecard-test:v1.10.0
    labels:
      suite: <test-suite>
      test: <test-name>
    storage:      
      spec:        
        mountPath:
          path: /test-output
```

Additionally if we can add a line at the beginning stating that users can enable local storage of scorecard results in either the CLI or in the config files.

Thanks.

Comment (michaelryanpeter) [reply modules/osdk-scorecard-store-output.adoc]: @theishshah I have made the requested changes along with a little bit of clean up. Would you PTAL?

**Review by theishshah — APPROVED** [review]
/lgtm

**Review by jmrodri — CHANGES_REQUESTED** [review]
/lgtm

**Review by jmrodri — APPROVED** [review]
/lgtm

File: modules/osdk-scorecard-store-output.adoc [hunk 2]
```diff
@@ -0,0 +1,51 @@
+// Module included in the following assemblies:
+//
+// * operators/operator_sdk/osdk-scorecard.adoc
+
+:osdk_ver: v1.16.0
+
+:_content-type: MODULE
```
Comment (kalexand-rh) [inline modules/osdk-scorecard-store-output.adoc:7]: Please pick a valid [content type](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#content-type-attributes).

File: modules/osdk-scorecard-store-output.adoc [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+[id="osdk-scorecard-store-output_{context}"]
+= Local storage of scorecard results
+
+You can enable local storage of scorecard test results by using the `scorecard` command or by modifying your `config.yaml` file.
```
Comment (kalexand-rh) [inline modules/osdk-scorecard-store-output.adoc:11]: This sounds like the abstract for a procedure module. I strongly suggest refactoring the content into discrete steps that follow the procedure template.

File: modules/osdk-scorecard-store-output.adoc [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+Running the `scorecard` command with the `--test-output` flag stores the scorecard test results in a specified directory. If you do not specify a directory, the command stores the test results in the `test-output` directory of your working directory.
+
+.`scorecard --test-output` command syntax
+[source,terminal,subs="attributes+"]
+----
+$ operator-sdk scorecard <bundle_directory_or_image> --test-output=/<output_directory> <.>
+----
+<.> Define the output directory.
+
+To enable local storage in your configuration files, you add the following to your `config.yaml` file:
```
Comment (kalexand-rh) [inline modules/osdk-scorecard-store-output.adoc:27]:
```suggestion
To enable local storage in your configuration files, you add the following <noun_goes_here> to your `config.yaml` file:
```

File: modules/osdk-scorecard-store-output.adoc [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+.Example scorecard configuration file
+[source,terminal,subs="attributes+"]
+----
+...
+tests:
```
Comment (kalexand-rh) [inline modules/osdk-scorecard-store-output.adoc:33]: Does it matter where you place this stanza?

File: modules/osdk-scorecard-store-output.adoc [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    image: quay.io/operator-framework/scorecard-test:v1.10.0
+    labels:
+      suite: <test_suite> <.>
+      test: <test_name> <.>
+    storage:
+      spec:
+        mountPath:
+          path: /<output_directory> <.>
+----
+<.> Replace with the command and arguments that you want to call, such as `scorecared-test`.
```
Comment (kalexand-rh) [inline modules/osdk-scorecard-store-output.adoc:46]:
```suggestion
<.> Replace `<test_entrypoint>` with the command and arguments to run, such as `scorecared-test`.
```
Will the user have enough context to define all of these parameters? If not, you might want to circle back and add more context.

File: modules/osdk-scorecard-store-output.adoc [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+<.> Define the test suite that you want to run, such as `basic` or `olm`.
```
Comment (kalexand-rh) [inline modules/osdk-scorecard-store-output.adoc:47]: Are these the only two options?

File: modules/osdk-scorecard-store-output.adoc [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+<.> Define the test name that you want to run, such as `olm-bundle-validation-test`.
+<.> Set the directory where you want to store your test results, such as `test-results`.
```
Comment (kalexand-rh) [inline modules/osdk-scorecard-store-output.adoc:49]:
```suggestion
<.> Set the location of the directory to store your test results, such as `test-results`.
```
Does this directory have to exist?

File: modules/osdk-scorecard-store-output.adoc [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+====
+
+Running the `scorecard` command with the `--test-output` flag stores the scorecard test results in a specified directory. If you do not specify a directory, the command stores the test results in the `test-output` directory of your working directory.
+
+.`scorecard --test-output` command syntax
+[source,terminal,subs="attributes+"]
+----
+$ operator-sdk scorecard <bundle_directory_or_image> --test-output=/<output_directory> <.>
+----
+<.> Define the output directory.
```
Comment (kalexand-rh) [inline modules/osdk-scorecard-store-output.adoc:25]: Honestly, I'd rewrite this whole thing without the broken flag and be like "this command store the test results in the `test-output` directory of your working directory." When the bug is fixed, update the docs to describe the flag.

**Review by kalexand-rh — COMMENTED** [review]
I have some suggestions and style/formatting requests.

File: modules/osdk-scorecard-store-output.adoc (hunk 9 above)
Comment (sferich888) [reply modules/osdk-scorecard-store-output.adoc:26]: +1 The why this all is written makes this read as if 'we have this cool new feature, but none of it works - good luck'. 

I wouldn't document this till the bug was/is fixed; only document the way's in which it properly works.

**Comment by bobfuru** [comment]
In preparation for OCP 4.10 GA, I'm moving the Milestone to "Next Release" as this PR did not make the merge by the 4.10 GA date. Post-GA, any open 4.10 PRs fall under the "Next Release" bucket in the same way that 4.6-4.9 PRs are already there. This change does not have any impact on the work in this PR; it's a housekeeping task to keep account of all PRs that had already merged by the 4.10 GA date. 😁

**Comment by bergerhoffer** [comment]
The `branch/enterprise-4.11` label has been added to this PR.

This is because your PR targets the `main` branch and is labeled for enterprise-4.10. And any PR going into `main` must also target the latest version branch (enterprise-4.11).

If the update in your PR does NOT apply to version 4.11 onward, please retarget this PR to go directly into the appropriate version branch or branches (enterprise-4.x) instead of main.

**Comment by bergerhoffer** [comment]
The enterprise-4.12 label has been added to this PR.

This is because your PR targets the `main` branch and is labeled for enterprise-4.11. And any PR going into `main` must also target the latest version branch (enterprise-4.12).

If the update in your PR does NOT apply to version 4.12 onward, please re-target this PR to go directly into the appropriate version branch or branches (enterprise-4.x) instead of main.

**Pull request closed without merging**